service/src/main/java/bio/terra/workspace/app/controller/ControlledGcpResourceApiController.java

```diff
@@ -541,6 +532,22 @@ private ApiCloneControlledGcpBigQueryDatasetResult fetchCloneBigQueryDatasetResu
         result, ControllerUtils.getAsyncResponseCode(result.getJobReport()));
   }
 
+  /**
+   * Extract the assigned user from a request to create a controlled resource. This field is only
+   * populated for private resources, but if a resource is private then a "null" value means "assign
+   * this resource to the requesting user" and should be populated here.
+   */
+  private String assignedUserFromBodyOrToken(
+      ApiControlledResourceCommonFields commonFields, AuthenticatedUserRequest userRequest) {
```

Comment:
from a law of demeter perspective, it's a little odd to pass in the containing object here instead of just passing in the privateResourceUser.

Reply:
This method is also evaluating the access scope from `commonFields`, since non-private resources don't have an assigned user. It either has to live inside this method or else everywhere it's called.
This could be two arguments instead. (accessScope + privateResourceUser) - I'd prefer to use the combined structure because I think it's a clearer interface and this has limited scope as a private method, but agreed it does make this a little less reusable.